river/neighbors/__init__.py

```diff
@@ -14,4 +15,5 @@
     "NearestNeighbors",
     "KNNClassifier",
     "KNNRegressor",
+    "ANNClassifier",
```

Comment:
By the way, what about `ANNRegressor`?

Reply:
How do you feel about having multiple possible "backends" in the future to power up the approximate search? The choice of SWINN or any other search index could become a parameter. I thought about leaving a generic name because of that.

I am also up for standardizing the names. NearestNeighbors is meant to be a generic class, just like SWINN. I decided not to expose SWINN for now, but we can also change that. Anyway, since they are generics, I think it is okay to have the full name (NearestNeighbors) or even rename it to LinearScan, ExhaustiveSearch, or something like that. What do you think?

Reply:
About the regressor, it is my next stop :D

Reply:
> LinearScan, ExhaustiveSearch, or something like that

Or BruteForceSearch

Reply:
> Ah yes that's a good idea. But so then would we have a single KNNClassifier and KNNRegressor, and pass it a parameter which would be the KNN algorithm? I like that actually. But it needs proper documentation.

I like this idea either, but documentation is a must. Would we pass either a `SWINN` or `NearestNeighbors`object to be used as the main buffering strategy?